sw/device/lib/dif/dif_adc_ctrl.c

```diff
@@ -403,3 +403,95 @@ dif_result_t dif_adc_ctrl_reset(const dif_adc_ctrl_t *adc_ctrl) {
 
   return kDifOk;
 }
+
+dif_result_t dif_adc_ctrl_irq_get_causes(const dif_adc_ctrl_t *adc_ctrl,
+                                         uint32_t *causes) {
+  if (adc_ctrl == NULL || causes == NULL) {
+    return kDifBadArg;
+  }
+
+  *causes = mmio_region_read32(adc_ctrl->base_addr,
+                               ADC_CTRL_ADC_INTR_STATUS_REG_OFFSET);
```

Comment:
More for the hardware design: This seems a bit funky. Was there a reason why these didn't get in the usual INTR_STATE CSR?

Reply:
yeah, @tjaychen and I chatted about this. Right now, the ADC Controller only generates a single interrupt, the `debug_cable` interrupt. However, there can be multiple "causes" for this interrupt, i.e. 1--8 filter matches or a sample being ready in `oneshot` mode. This does not quite align with how other blocks generate IRQs, and we will most likely reorganize IRQs in the future to be more consistent across blocks, my understanding was this just isn't super high priority right now? Currently there are autogenerated ISR testutils (#9038), so I can update the autogen'd code to accommodate this extra step of "clearing" the IRQ from this block util the HW can be adjusted to make this more consistent with other blocks.